### ISSUE TITLE
Similar(X::ACSet)

### ISSUE THREAD

**Issue by KevinArlin** [AlgebraicJulia/ACSets.jl#65]
There are some places in the code (for instance in the nauty integration, but I bet not just there) where we copy or even `deepcopy` a whole acset just to get an acset of the same shape that we'll then overwrite. It might be nice to write a method of `Base.similar` that copies an acset's shape (i.e. attrtypes and cardinalities of homs) but leaves everything uninitialized for this kind of situation. I don't know enough about columns to know whether this would be trivial or annoying.

**Comment by epatters** [comment]
Good idea. It should be straightforward to implement this feature without reaching into the internals. Just create an empty acset of the same type and call `add_parts!` without setting any subparts.
